docs/user/urls.md

```diff
@@ -0,0 +1,117 @@
+---
+sidebar_position: 10
+---
+
+# Agama URLs
+
+When you want to perform an unattended installation, Agama needs to know where
+to find the profile. Usually, you make use of the `inst.auto` argument, which
+receives a URL. Agama supports a wide range of protocols, including
+`http(s)://`. `ftp://`, `file://`, etc. However, it supports most of the
+[AutoYaST specific URLs](https://doc.opensuse.org/projects/autoyast/#Commandline-ay)
+too.
+
+## Supported schemes
+
+Agama supports the following standard schemes[^1]:
+
+- `http` and `https` for HTTP and HTTPS.
+- `ftp` and `ftps` for FTP and FTP over TSL.
+- `smb` and `smbs` for the Server Message Block (SMB) protocol[^2].
+- `nfs` for the Network File System (NFS) protocol.
+- `file` for local files.
+
+In addition to those schemes, Agama supports a set of specific ones which can be
+rather useful:
+
+- [`device`](#device) for finding files in specific devices (even unmounted
+  ones).
+- [`usb`](#usb) for finding files in USB devices (even unmounted ones).
+- [`label`](#label) for finding files in a file system with a given label.
+- [`cd`, `dvd`, `hd`](#cd-dvd-and-hd) for finding files in specific devices
+  (prefer `device`).
+
+## Agama specific schemes
+
+The typical protocols are enough for most use cases. However, more specific URLs
+can be handy in many situations. For instance, instead of serving your profile
+through HTTP, you could just put it on a USB stick and tell Agama about it with
+`usb:///profile.json`. Agama would search in all your USB devices for the
+profile to use. Cool, right?
+
+Let's have a look to the different kind of URLs that Agama offers.
+
+### `device`
+
+This is the most flexible scheme. The general form is `device://[DEVICE]/PATH`.
+Let's have a look to some examples to better understand of how it works.
+
+In the following example, Agama will search for the `profile.json` file in the
+`/dev/sda1` device:
+
+```text
+device://sda1/profile.json
+```
+
+It is possible to search in a given directory. In this case, the
+`/profiles/sles.json` file in the `/dev/sda1` device:
+
+```text
+device://sda1/profiles/sles.json
+```
+
+Nested devices are supported too:
```

Comment:
I have never heard of such device. Do you have a link to Wikipedia / openSUSE.org?

Reply:
Well, I found them in the YaST code: https://github.com/openSUSE/microos-tools/blob/fa1a910b31274675516b29443923738d3738bba2/NEWS#L2.

Reply:
I meant https://github.com/yast/yast-installation/blob/master/src/lib/transfer/file_from_url.rb#L483-L486.